pkg/resource/deploy/providers/provider.go

```diff
@@ -41,22 +61,31 @@ type ProviderRequest struct {
 	pkg               tokens.Package
 	pluginDownloadURL string
 	pluginChecksums   map[string][]byte
+	parameterization  *ProviderParameter
```

Comment:
Anything we should add to the doc comment around parameterisation here? Reading it, it looks like it wouldn't affect `ProviderRequest`'s usefulness as a hash key -- is that correct?